src/mds/CInode.cc

```diff
@@ -1800,6 +1800,7 @@ void CInode::encode_lock_ifile(bufferlist& bl)
       encode(get_inode()->truncate_size, bl); 
       encode(get_inode()->client_ranges, bl); 
       encode(get_inode()->inline_data, bl); 
+      encode_raw(get_inode()->fscrypt_priv, bl);
```

Comment:
A test case to ensure this works properly: changing the encrypted size but **not** changing the CephFS size correctly locks/unlocks the `ifile` lock. Presumably, the kernel won't be sending a redundant update to the size if it does not change but the encrypted size might change.